packages/cli/src/cli/cogify/batch.job.ts

```diff
@@ -111,24 +119,43 @@ export class BatchJob {
     fsa.configure(job.output.location);
     const runningJobs = await BatchJob.getCurrentJobList(batch);
 
+    const chunkJob: string[] = [];
     const stats = await Promise.all(
-      job.output.files.map(async ({ name }) => {
-        if (oneCog != null && oneCog !== name) return { name, ok: true };
-        const jobName = BatchJob.id(job, name);
+      job.output.files.map(async ({ name, width }) => {
+        // Check existence of the output.
+        const targetPath = job.getJobPath(`${name}.tiff`);
+        const exists = await fsa.exists(targetPath);
+        if (exists) {
+          logger.info({ targetPath }, 'FileExists');
+          return { names: [name], ok: true };
+        }
+
+        // Calculate the imagery size and chunk the small ones into one job.
+        const size = width / job.output.gsd;
+        if (size < ChunkJobSizeLimit) chunkJob.push(name);
+
+        // Check existence of the job running.
+        const jobName = BatchJob.id(job, [name]);
```

Comment:
I'm not sure this logic is going to work, I think you need to create a list of jobs first. then iterate to see if the jobs exist and then submit them.

as the chunked job isnt going to have the same name as `BatchJob.id(job, [name])` vs `BatchJob.id(job, [a,b,c])`